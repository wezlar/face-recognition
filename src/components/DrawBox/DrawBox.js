import React from 'react';
import PropTypes from 'prop-types';
import DrawNameBox from '../DrawNameBox/DrawNameBox';

const boxStyle = {
  position: 'absolute',
  border: 'solid',
  borderColor: 'blue',
};

export const DrawBox = (props = {}) => {
  const { detections, match } = props;
  let drawBox = null;

  if (!!detections) {
    drawBox = detections.map((detection, i) => {
      let _H = detection.box.height;
      let _W = detection.box.width;
      let _X = detection.box._x;
      let _Y = detection.box._y;

      return (
        <div key={i}>
          <div
            style={{
              ...boxStyle,
              height: _H,
              width: _W,
              transform: `translate(${_X}px, ${_Y}px)`,
            }}>
              <DrawNameBox 
                match={match} 
                index={i} 
                width={_W} 
                height={_H} />
            </div>
        </div>
      );
    });
  }

  return drawBox;
};

DrawBox.propTypes = {
  detections: PropTypes.array,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]),
  match: PropTypes.array,
};

export default DrawBox;
