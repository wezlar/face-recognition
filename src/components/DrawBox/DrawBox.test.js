import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';

import DrawBox from './DrawBox';

afterEach(cleanup);

describe(`Test DrawBox`, () => {
  test(`Test snapshot`, () => {
    const match = [
      {
        _label: 'Test Label 1',
      },
      {
        _label: 'Test Label 2',
      }
    ];

    const detections = [
      {
        box: {
          height: 100,
          width: 200,
          _x: 300,
          _y: 400,
        },
      },
    ];

    const { container, getByTestId, debug } = render(
      <DrawBox 
        match={match} 
        detections={detections} />
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('draw-box').children.length).toBe(1);
    expect(getByTestId('draw-box').style.height).toBe('100px');
    expect(getByTestId('draw-box').style.width).toBe('200px');
    expect(getByTestId('draw-box').style.transform).toBe('translate(300px, 400px)');
  });
});