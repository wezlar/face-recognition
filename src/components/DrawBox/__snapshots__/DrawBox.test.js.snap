// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DrawBox Test snapshot 1`] = `
<div
  data-testid="draw-box"
  style="position: absolute; border: solid; border-color: blue; height: 100px; width: 200px; transform: translate(300px, 400px);"
>
  <p
    data-testid="draw-name-box"
    style="background-color: blue; border: solid; border-color: blue; margin-top: 0px; color: rgb(255, 255, 255); width: 200px; transform: translate(-3px, 100px);"
  >
    Test Label 1
  </p>
</div>
`;
